polygeist/lib/Dialect/Polygeist/Transforms/LICM.cpp

```diff
@@ -703,16 +703,44 @@ void AffineParallelGuardBuilder::getConstraints(
     }
 }
 
+//===----------------------------------------------------------------------===//
+// LICMCandidate
+//===----------------------------------------------------------------------===//
+
+class LICMCandidate {
+public:
+  LICMCandidate(Operation &op) : op(op) {}
+  Operation &getOp() const { return op; }
+  void addDependee(Operation &op) { dependees.push_back(&op); }
+
+private:
+  Operation &op;
+  SmallVector<Operation *> dependees;
+};
+
 //===----------------------------------------------------------------------===//
 
+/// Check if \p willBeMoved contains \p op. If true, add \p op to \p candidate
+/// list of dependees, and return true. Otherwise, return false.
```

Comment:
If `op` is an operation that has already been proven invariant (and therefore is in `willBeMoved`) what makes it a `dependee` of `candidate` ? To understand one requires the content and given this function is quite small I prefer to `inline` it.